src/screens/DiscoveryScreenV2/index.js

```diff
@@ -61,63 +68,120 @@ const DiscoveryScreenV2 = ({route}) => {
+        setDiscoveryDataUnfollowedUsers(
+          data?.unfollowedUsers?.map((item) => ({
+            ...item,
+            following: item.user_id_follower !== null
+          })) || []
+        );
       }
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         user: false
       }));
-    });
+    };
+
+    const fetchDiscoveryInitialDomains = async () => {
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryDomains();
+      FollowingAction.setFollowingDomain(initialData.suggestedDomains, followingDispatch);
+      DiscoveryAction.setDiscoveryInitialDomains(initialData.suggestedDomains, discoveryDispatch);
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataDomain(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryDataDomain = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataDomain(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataFollowedDomains(data?.followedDomains || []);
         setDiscoveryDataUnfollowedDomains(data?.unfollowedDomains || []);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         domain: false
       }));
-    });
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataTopic(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryInitialTopics = async () => {
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryTopics();
+      DiscoveryAction.setDiscoveryInitialTopics(initialData.suggestedTopics, discoveryDispatch);
+    };
+
+    const fetchDiscoveryDataTopic = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataTopic(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataFollowedTopics(data?.followedTopic);
         setDiscoveryDataUnfollowedTopics(data?.unfollowedTopic);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         topic: false
       }));
-    });
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataNews(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryDataNews = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataNews(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataNews(data?.news);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         news: false
       }));
-    });
+    };
+
+    if (text === undefined) {
+      await Promise.all([
+        fetchDiscoveryInitialUsers(),
+        fetchDiscoveryInitialDomains(),
+        fetchDiscoveryInitialTopics(),
+        fetchDiscoveryDataNews()
+      ]);
+    } else {
+      await Promise.all([
+        fetchDiscoveryDataUser(),
+        fetchDiscoveryDataDomain(),
+        fetchDiscoveryDataTopic(),
+        fetchDiscoveryDataNews()
+      ]);
+    }
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This is a significant refactor of the `fetchDiscoveryData` function. The new version appears to be more modular and easier to read than the old one. It separates the fetching of different types of data into separate functions, which can improve maintainability. However, there are no error handling mechanisms in these async functions. If any of the API calls fail, the error will not be caught and handled properly. Consider adding try-catch blocks around the API calls to handle potential errors.

```diff
    const fetchDiscoveryDataUser = async () => {
      const cancelToken = cancelTokenRef?.current?.token;
      try {
        const data = await DiscoveryRepo.fetchDiscoveryDataUser(text, {cancelToken});
        if (data.success) {
          setDiscoveryDataFollowedUsers(
            data?.followedUsers?.map((item) => ({
              ...item,
              following: item.user_id_follower !== null
            })) || []
          );
          setDiscoveryDataUnfollowedUsers(
            data?.unfollowedUsers?.map((item) => ({
              ...item,
              following: item.user_id_follower !== null
            })) || []
          );
        }
      } catch (error) {
        console.error(error);
      }
      setIsLoadingDiscovery((prevState) => ({
        ...prevState,
        user: false
      }));
   };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
it's already handled in services level